src/graphiql.js

```diff
@@ -0,0 +1,118 @@
+import React, { useRef, useEffect, useState } from 'react';
+import GraphiQL from 'graphiql';
+import PropTypes from 'prop-types';
+import { buildSchema } from 'graphql';
+import GraphiQLExplorer from 'graphiql-explorer';
+import './common/styles/index.scss';
+
+const TykGraphiQL = (props) => {
+  const {
+    fetcher,
+    schema,
+    ideName = 'TykGraphiQL',
+    query: initialQuery = '{}',
+    showPlayground = true,
+    menu,
+  } = props;
+  const playgroundWrapperRef = useRef(null);
+  let playgroundRef = useRef(null);
+  const [explorerIsOpen, setExplorerIsOpen] = useState(false);
+  const [query, setQuery] = useState(initialQuery);
+  const [selectedMenu, setSelectedMenu] = useState(menu?.title);
+
+  useEffect(() => {
+    if (!playgroundWrapperRef.current) {
+      return;
+    }
+
+    playgroundWrapperRef.current.addEventListener('click', (e) => {
+      e.preventDefault();
+    });
+  }, [playgroundWrapperRef]);
+
+  const toggleExplorer = () => setExplorerIsOpen(!explorerIsOpen);
+
+  const handleMenuSelection = (item) => {
+    menu.onSelect(item);
+    setSelectedMenu(item?.label);
+  };
+
+  return (
+    <div
+      ref={playgroundWrapperRef}
+      className="graphiql-container"
+      style={{
+        visibility: showPlayground ? 'visible' : 'hidden',
+      }}
+    >
+      <GraphiQLExplorer
+        explorerIsOpen={explorerIsOpen}
+        onToggleExplorer={toggleExplorer}
+        onEdit={(q) => setQuery(q)}
+        schema={schema && buildSchema(schema)}
+        query={query}
+        onRunOperation={(operationName) =>
+          playgroundRef.handleRunQuery(operationName)
+        }
+      />
+      <GraphiQL
+        ref={(ref) => {
+          playgroundRef = ref;
+          return true;
+        }}
```

Comment:
This ref function seems weird to me.
Not sure what `return true` is needed for.
Also usually you would do something like `ref={playgroundRef}`. Or `ref={(element) => playgroundRef.current = element}`.
But anyway, if it works like this I won't ask too many questions :)

Reply:
` return true`
was to avoid linting errors related to arrow fn